server/leafnode.go

```diff
@@ -116,19 +116,22 @@ func (s *Server) addInJSDeny(r *leafNodeCfg) {
 	s.addInJSDenyImport(r)
 }
 
+// To deny all JS traffic.
+const jsDenyAll = "$JS.>"
+
 // Will add in the deny export for JetStream on solicited connections if we
 // detect we have multiple JetStream domains and we know our local account
 // is JetStream enabled.
 // r lock should be held.
 func (s *Server) addInJSDenyExport(r *leafNodeCfg) {
 	for _, dsubj := range r.DenyExports {
-		if dsubj == jsAllAPI {
+		if dsubj == jsDenyAll {
 			return
 		}
 	}
 
-	s.Noticef("Adding deny export of %q for leafnode configuration on %q that bridges system account", jsAllAPI, r.LocalAccount)
-	r.DenyExports = append(r.DenyExports, jsAllAPI)
+	s.Noticef("Adding deny export of %q for leafnode configuration on %q that bridges system account", jsDenyAll, r.LocalAccount)
+	r.DenyExports = append(r.DenyExports, jsDenyAll)
```

Comment:
Because `$JS.>` covers the result of our mapping, I would not expected this to work.
My assumption is that because we deny `$JS.API.>` here, the result of the mapping `$JS.<domain>.API...` would not be denied.
Now we deny `$JS.>` but our mapping from $JS.<domain>.. is covered by it.

 
My expectation would be to see something like this instead. 
r.DenyExports = append(r.DenyExports, jsAllAPI)
r.DenyExports = append(r.DenyExports, jsAckAPI)
r.DenyExports = append(r.DenyExports, jsFlowControlAPI)